src/table/styling-utils.js

```diff
@@ -13,12 +14,84 @@ export function getHeadStyle(layout, theme) {
+    b = parseInt(matches[3], 16);
+  }
+  // Using the HSP (Highly Sensitive Poo) value, determine whether the color is light or dark
+  // HSP < 125, the color is dark, otherwise, the color is light
+  return 0.299 * r + 0.587 * g + 0.114 * b < 125;
+}
+
+function isUnset(prop) {
+  return !prop || JSON.stringify(prop) === JSON.stringify({ index: -1, color: null });
+}
+
 export function getBodyStyle(layout, theme) {
   const content = layout.components?.[0]?.content;
-  return content
-    ? {
-        fontColor: getColor(content.fontColor, '#404040', theme),
-        fontSize: content.fontSize || '14px',
-      }
-    : {};
+  if (!content) return {};
+
+  //  "hoverFontColor": {
+  //   "index": -1,
+  //   "color": null
+  // } means the hover fornt color is unset.
+  const unsetFontColor = isUnset(content.fontColor);
+  const unsetHoverFontColor = isUnset(content.hoverFontColor);
+  const unsetHoverBackgroundColor = isUnset(content.hoverColor);
+  const unsetHoverFontandBackgroundColor = unsetHoverFontColor && unsetHoverBackgroundColor;
+
+  // 1. hoverEffect is true, there is no font color and no hover font color but a hover background color,
+  // when hovering, the hover font color becomes white when the hover background color is a dark color
+  // or the hover font color becomes black when the hover background color is a light color.
+
+  // 2. hoverEffect is true, there is no hover font color but a font color and a hover background color ,
+  // when hovering, the hover font color stays the font color if the hover background color is light,
+  // or the hover font color becomes white if the hover background color is dark.
+
+  // 3. hoverEffect is true, there is a hover font color but no hover background color,
+  // when hovering, only a hover font color is applied when hovering and and the hover background color disappears.
+
+  // 4. hoverEffect is true, there is a font color but no hover font color and no hover background color, when hovering, the defalut hover effect
+
+  // 4. hoverEffect is true, there are all colors, when hovering, the set hover font color and the the hover background color take effect.
+
+  const getHoverBackgroundColor = unsetHoverFontandBackgroundColor
+    ? 'rgba(0, 0, 0, 0.03)'
+    : unsetHoverBackgroundColor
+    ? 'rgba(0, 0, 0, 0)'
+    : getColor(content.hoverColor, 'rgba(0, 0, 0, 0.03)', theme);
+
+  const getHoverFontColor = unsetHoverFontandBackgroundColor
+    ? ''
+    : unsetFontColor
+    ? isDarkColor(getHoverBackgroundColor)
+      ? getColor(content.hoverFontColor, '#ffffff', theme)
+      : getColor(content.hoverFontColor, '#000000', theme)
+    : unsetHoverFontColor
+    ? isDarkColor(getHoverBackgroundColor)
+      ? '#ffffff'
+      : ''
+    : getColor(content.hoverFontColor, '', theme);
+
+  return {
+    fontColor: getColor(content.fontColor, '#404040', theme),
+    fontSize: content.fontSize || '14px',
+    hoverBackgroundColor: `${getHoverBackgroundColor} !important`,
```

Comment:
important is not needed. The selection background will still override the hover background. This is the way it is in the old table, so let's keep it like that for now at least

Reply:
important is necessary. Otherwise, in sense-client, the hover effect skips every other row.

Reply:
I have a fix for it in sense-client. It is a weird css rule for the old table that is applied to sn-table as well: https://github.com/qlik-trial/sense-client/pull/15161

Reply:
Nice.